symplyphysics/laws/thermodynamics/work_of_ideal_gas_in_isothermal_process.py

```diff
@@ -30,9 +43,39 @@
 law = Eq(work, (gas_mass / molar_mass) * units.molar_gas_constant * temperature *
     log(final_volume / start_volume))
 
+# Derive from ideal gas equation
 
-def print_law() -> str:
-    return print_expression(law)
+_mole_count = solve(
+    molar_qty_law.law, molar_qty_law.amount_of_substance,
+)[0].subs({
+    molar_qty_law.extensive_quantity: gas_mass,
+    molar_qty_law.molar_quantity: molar_mass,
+})
+
+_ideal_gas_eqn = ideal_gas_equation.law.subs({
+    ideal_gas_equation.mole_count: _mole_count,
+    ideal_gas_equation.temperature: temperature,
+})
+
+_pressure_expr = solve(_ideal_gas_eqn, ideal_gas_equation.pressure)[0]
+
+_volume_before = SymSymbol("volume_before", positive=True)
+_volume_after = SymSymbol("volume_after", positive=True)
+
+_work_expr = work_law.law.rhs.xreplace({
+    work_law.volume: ideal_gas_equation.volume,
```

Comment:
Use inverse replacement - substitute `ideal_gas_equation.volume` for `work_law.volume`

Reply:
That's a good solution.